### PR TITLE
CI: Fix matplolib release issues

### PR DESCRIPTION
**Pull request by phofl**
- [ ] xref #48602 (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I am not really familiar with matplotlib, so I've xfailed where the fix was not clear to me

Also gets rid of deprecation warnings

### CODE REVIEW

File: pandas/tests/plotting/common.py [hunk 1]
```diff
@@ -598,3 +598,16 @@ def _gen_two_subplots(f, fig, **kwargs):
     else:
         kwargs["ax"] = fig.add_subplot(212)
     yield f(**kwargs)
+
+
+try:
+    import matplotlib as mpl
```
Comment (lithomas1) [inline pandas/tests/plotting/common.py:604]: You probably want to add 3.6.0 to https://github.com/pandas-dev/pandas/blob/main/pandas/plotting/_matplotlib/compat.py#L20-L21 instead.

Comment (phofl) [reply pandas/tests/plotting/common.py:604]: Thx, changed

**Comment by phofl** [comment]
cc @datapythonista not sure if this is relevant for the release

**Review by lithomas1 — APPROVED** [review]

**Comment by phofl** [comment]
Thx, I am merging for now to get ci back to green. Please ping if there is something we could improve here.

**Pull request merged**